### ISSUE TITLE
Stacking Tracker

### ISSUE THREAD

**Issue by will-corcoran** [stacksgov/critical-bounties#28]
**Recipient:** Philip.btc
**Bounty:** $51,500

### Summarized

- **Background**
    - Aim: Provide in-depth data on proof of transfer (PoX) and replace the defunct stacking.club.
    - Importance: Serves the mission of a user-owned internet by offering detailed analytics and insights on Stacks infrastructure.
- **Project Overview**
    - Goal: Build a comprehensive application for Stacks infrastructure, including Stacking/PoX, Signer Tools & Data, and Miner data.
- **Scope**
    - Initial Scope: Key functionalities for the first release, focusing on user stories (see below) and cost.
    - Deliverables:
        1. Authenticate with the Leather Wallet.
        2. View basic information on pools, APY, rewards, TVL.
        3. View pool stacking and rewards data.
        4. View historical rewards with filters.
        5. View PoX cycles and payouts.
        6. Compare different LSTs.
        7. View signers and delegated STX.
        8. Install TG bot for reward notifications.
- **Team**
    - Experience: Developed Arkadiko and Stacking DAO, with 10+ years in startups and tech.
- **Risks**
    - Main Challenge: Ensuring accurate and easy-to-interpret data for users.

### User Stories

- **User Story**: As a user, I can authenticate with the app through the Leather Wallet and see my PoX data.
- **User Story**: As a user or guest, I can view basic information on the landing page, such as different pools, APY, total stacking rewards, TVL.
- **User Story**: As a user or guest, I can see the pools in the ecosystem, how much they are stacking and how many rewards they are paying.
- **User Story**: As a user or guest, I can see how many rewards were paid last cycle or over time (with custom filters).
- **User Story**: As a user or guest, I can historically view PoX cycles and see how many STX were stacked per cycle, including payouts.
- **User Story**: As a user or guest, I can see the different LSTs on Stacks and how they compare to each other.
- **User Story**: As a user or guest, I can see the different signers in the ecosystem, whether they run a pool and how much STX is delegated to them.
- **User Story**: As a user, I can install the TG bot to receive notifications on new rewards per pool.

## Deliverables

1. **Deliverable 1**: Authentication with the Leather Wallet.
2. **Deliverable 2**: View basic information on pools, APY, rewards, TVL.
3. **Deliverable 3**: View pool stacking and rewards data.
4. **Deliverable 4**: View historical rewards with filters.
5. **Deliverable 5**: View PoX cycles and payouts.
6. **Deliverable 6**: Compare different LSTs.
7. **Deliverable 7**: View signers and delegated STX.
8. **Deliverable 8**: Install TG bot for reward notifications.

## Roadmap & Timeline

- **Phase 1**: Initial development and setup (4 weeks).
    - Authenticate with the Leather Wallet.
    - View basic information on pools, APY, rewards, TVL.
- **Phase 2**: Development of core features (6 weeks).
    - View pool stacking and rewards data.
    - View historical rewards with filters.
    - View PoX cycles and payouts.
- **Phase 3**: Advanced features and integrations (6 weeks).
    - Compare different LSTs.
    - View signers and delegated STX.
    - Install TG bot for reward notifications.
- **Phase 4**: Testing, refinement, and launch (4 weeks).
    - Conduct comprehensive testing and refinement.
    - Finalize and launch the application.

**Comment by tycho1212** [comment]
🔥🔥🔥

**Comment by philiphacks** [comment]
🚀

**Comment by will-corcoran** [comment]
Hi @philiphacks -

I hope you are well.  

Now that we are about a month into having awarded this Critical Bounty, we would kindly request that you provide an update here for the community.  Before the end of the week (Friday, August 16th) can you please provide us an update on the status of your Critical Bounty, please make sure to include:
- Summary of CB-related work that has been done to date
- Summary of CB roadmap from here to completion
- Flag an potential blockers
- Anticipated date of completion
- Link to all related repos (remember all funded work should be open source)

Thanks!
Will

cc: @shaktistacks

**Comment by philiphacks** [comment]
Hi @will-corcoran - thanks for pinging!

We've been hard at work on this in July and August, I'm traveling today until Sunday but I'll make a comprehensive overview of our progress (including open sourcing the repo of our work we already did) early next week, apologies for the delay on this.

We regularly deploy progress to https://www.stacking-tracker.com/ fyi (note: this is not designed yet and just raw data, of which some might be incorrect)

**Comment by philiphacks** [comment]
Here's an update on the user stories/deliverables:

# Deliverables
## Deliverable 1: Authentication with the Leather Wallet.
Done

## Deliverable 2: View basic information on pools, APY, rewards, TVL.
Done

## Deliverable 3: View pool stacking and rewards data.
Done

## Deliverable 4: View historical rewards with filters.
TODO

## Deliverable 5: View PoX cycles and payouts.
Done

## Deliverable 6: Compare different LSTs.
TODO

## Deliverable 7: View signers and delegated STX.
Done

## Deliverable 8: Install TG bot for reward notifications.
TODO

The next deliverable we will be working on is 4 and 6 (ETA end of September). After that we will take some time to get everything ready for production and design it to be more user friendly (ETA end of October). In parallel, we can build a TG bot (ETA end of October).

There are no blockers. A production-ready version of the Stacking Tracker, which could eventually be hosted on stacking.club too, will be available around the end of October.

I hope that helps! I'll drop a link to the repo too, just checking if we're all good to open source it (e.g. no sensitive data is in it)

**Comment by philiphacks** [comment]
@will-corcoran see public repo here: https://github.com/StackingDAO/stacking-tracker

**Comment by will-corcoran** [comment]
Hi @philiphacks 

 We're exploring allocating our remaining 2024 grants budget and considering follow-up funding for current Critical Bounty recipients. If you have ideas to enhance your previously awarded work, we'd like to hear from you.

### **Next Steps:**

1. If you're interested in follow-up funding, please let us know by the end of this week.
2. You can either:
  -   Post your proposal as a comment on this issue (preferred for final submissions)
  -   Email me your initial thoughts if you're still developing your idea or prefer privacy

### **Proposal Details:**

- Please include:
  - Additional scope or enhancements
  - Associated costs
  - Justification for how a follow-up grant could significantly boost the ROI of the initial grant

We look forward to your ideas!

Best,
Will

**Comment by philiphacks** [comment]
@will-corcoran hey will, thanks for reaching out!

I think at this time we won't need additional funding for this. We are progressing really well and I think the current grant allocation works to launch something great and useful.

We will continue to maintain this stacking tracker internally within Stacking DAO as well, but perhaps if we see a clear need for a grant, we might apply for an additional one in the future (but not this year)

**Comment by will-corcoran** [comment]
Hey @philiphacks just wanted to reach out and see how things are going?  Have you completed all of your milestones?  Can you please provide an update?  Thanks!

**Comment by philiphacks** [comment]
@will-corcoran yes, we have finished up the logic in October and have spent the last 3 weeks on a redesign (complete UI overhaul) to make it look pretty (& better UX) for launch in 2 weeks.

Let's hop on a call for a demo, and then I'll drop a longer overview of what was built here too!

**Comment by friedger** [comment]
Nice work!

My feature requests :
* the historical values of minium stx for 1 slot. 
* the btc/stx price when showing the btc rewards.
* an explanation how the APY is calculated.

Issues:
* I see an API request to /balances with a testnet address when connection my wallet on "My position". Nothing is shown to me when connected.
* I see 45 pools (private and public) that have used aggregate commit for cycle 98.
* On https://www.stacking-tracker.com/signers/fast-pool the pending APY is not shown while it is shown on other pages.

**Comment by MarvinJanssen** [comment]
I echo @friedger but very nice work @philiphacks!

<img width="916" alt="image" src="https://github.com/user-attachments/assets/4a4339e4-bcb3-48a4-8c26-2d9c21317e30">

Maybe?

**Comment by friedger** [comment]
Source code is not available anymore at https://github.com/StackingDAO/stacking-tracker
